department-of-veterans-affairs/caseflow
# changing deprecated functions to componentdidupdate

Resolves [APPEALS-4961](https://jira.devops.va.gov/browse/APPEALS-4961)

# Description
Remove unsafe_componentWillMount, unsafe_componentwillreceiveProps, unsafe_componentWillUpdate and changed it to componentdidUpdate. 
Added a snapshot for the jest test.
Added some propsType validation for linter

## Acceptance Criteria
- [ ] All the feature should work properly after removal of unsafe react cycle
- [ ] Search Bar should be functioning as expected
- [ ] Searchable Dropdown should be working as expected.

## Testing Plan

1. Go to [Jira Issue/Test Plan Link](https://jira.devops.va.gov/browse/JIRA-12345) or list them below
2. When switching users -> Searchable dropdown at UserSelector.
3.  Switch User to ACBAUERVVHAH
4. Go to /intake
5.  SELECT HLR in Whichform are you processing
6.  Insert a Veteran's ID that is present in the system 
7.  fill all the form, for Benefit type select -> Compensation and for Is the Claimant someone other than the veteran? Select Yes.
8.  This should make 'Please Select claimant listed on the form' radio to be visible. Select Bob and expect payee code for this claimant to be autopopulated to 10-Spouse, same with other option they should be auto populated. 
9.  Login as any reader user, 
10. Generate Document and see if selectorDropdown is working as expected in document.


# Frontend
## User Facing Changes

 BEFORE|AFTER
 ---|---

## Storybook Story
*For Frontend (Presentation) Components*

# Backend
## Database Changes
*Only for Schema Changes*


## Integrations: Adding endpoints for external APIs

# Best practices
## Code Documentation Updates

## Tests
### Test Coverage
Did you include any test coverage for your code? Check below:

### Code Climate
Your code does not add any new code climate offenses? If so why?

## Monitoring, Logging, Auditing, Error, and Exception Handling Checklist

### Monitoring

### Logging

### Auditing

### Error Handling

### Exception Handling
